### PR TITLE
made dropout keep_prob keyword arg consistent with tensorflow

### LINKED ISSUE #6: DOC dropout documentation is confusing
It appears that the dropout documentation is a bit confusing. The parameter is named "dropout" but is fed to the "keep_prob" parameter in the TF dropout function. Either the docs should be changed or maybe `1 - dropout` should be fed to the `keep_prob`.

### PR DESCRIPTION
**Pull request by mheilman**
closes #6 (applies to both the MLP and autoencoder).

This changes the `dropout` keyword argument to `keep_prob` to be consistent internally and consistent with tensorflow.

### CODE REVIEW

**Comment by mheilman** [comment]
@beckermr, look OK?

**Comment by beckermr** [comment]
The code itself seems fine. However, I personally would prefer to just keep the keyword as `dropout` but feed `1.0 - self.dropout` to `tensorflow` as `keep_prob`. This is clearer to the user and matches what is in the autoencoder. Thoughts?

**Comment by mheilman** [comment]
It's a bit arbitrary, but I think just calling the parameter `dropout` is confusing.  It sounds like it should be the probability of dropping a unit, but the [dropout paper](https://www.cs.toronto.edu/~hinton/absps/JMLRdropout.pdf) talks about having a paramater `p` that is the probability of keeping/retaining units (e.g., fig 2).

I'm guessing TF uses `keep_prob` to make the behavior more explicit, and I sort of think we should do the same.  Also, I think it's probably good to be consistent with TF's naming conventions

**Comment by beckermr** [comment]
Hmmm. This is fine I guess. I look at the keyword `keep_prob` and don't have any idea what it refers to. The `keep_prob` of what? Maybe `dropout_keep_prob`? In any case, we should def have one convention in the code base. Can you change the autoencoder to whatever we pick as well?

**Comment by mheilman** [comment]
Yeah, I thought about `dropout_keep_prob`. I had it as just `dropout` since it seems like sklearn often uses short, slightly ambiguous parameter names in these sorts of cases (e.g., `degree` in [SVMs](http://scikit-learn.org/stable/modules/generated/sklearn.svm.SVR.html), which only applies for polynomial kernels, instead of `polynomial_kernel_degree`).

I can change it to that if you'd like.

This PR already changes the autoencoder to use `keep_prob`, for consistency.  I've updated the first comment of this PR to make that clear since it's a bit more than just addressing #6.

**Comment by beckermr** [comment]
Ahh cool! OK fine. I retract my previous comment. `keep_prob` is fine.

File: muffnn/autoencoder/autoencoder.py [hunk 1]
```diff
@@ -176,8 +177,7 @@ def _set_up_graph(self):
         second_layers \
             = list(self.hidden_units[::-1][1:]) + [self.input_layer_size_]
         for i, layer_sz in enumerate(second_layers):
-            if self.dropout is not None:
```
Comment (beckermr) [inline muffnn/autoencoder/autoencoder.py]: This branch could be kept with a `keep_prob == 1.0` test.

Comment (mheilman) [inline muffnn/autoencoder/autoencoder.py]: yeah, true.  

At first, i thought that doing this wasn't necessary because of the check that TF has [here](https://github.com/tensorflow/tensorflow/blob/r0.11/tensorflow/python/ops/nn_ops.py#L1184), but after another look I think it's still needed since we have `keep_prob` as a placeholder.

Comment (mheilman) [inline muffnn/autoencoder/autoencoder.py]: it looks like [these objects](https://github.com/tensorflow/tensorflow/blob/r0.11/tensorflow/python/ops/nn_ops.py#L1187-L1196) still get added to the graph and are probably not trivial computationally even when 1.0 is fed in for `keep_prob`.

File: muffnn/autoencoder/autoencoder.py [hunk 2]
```diff
@@ -38,8 +38,10 @@ class Autoencoder(TFPicklingBase, TransformerMixin, BaseEstimator):
     n_epochs : int, optional
         The number of epochs (iterations through the training data) when
         fitting.
-    dropout : float or None, optional
-        The dropout probability. If None, then dropout will not be used.
+    keep_prob : float, optional
+        The probability of keeping values in dropout. A value of 1.0 means that
+        dropout will not be used. cf. `TensorFlow documentation
+        <https://www.tensorflow.org/versions/r0.11/api_docs/python/nn.html#dropout>`
```
Comment (mheilman) [inline muffnn/autoencoder/autoencoder.py]: please see my response to the comment below and let me know what you think.  if we add those checks for `keep_prob == 1.0` back in, then the current docstring should work, right?

File: muffnn/mlp/mlp_classifier.py [hunk 3]
```diff
@@ -97,7 +99,7 @@ def _init_model_output(self, t):
             t = affine(t, output_size, input_size=self.input_layer_sz_,
                        scope='output_layer', sparse_input=True)
         else:
-            t = tf.nn.dropout(t, keep_prob=self._dropout)
+            t = tf.nn.dropout(t, keep_prob=self._keep_prob)
```
Comment (beckermr) [inline muffnn/mlp/mlp_classifier.py]: Do you need to test for keep prob not equal to 1 here?

File: muffnn/mlp/mlp_regressor.py [hunk 4]
```diff
@@ -92,7 +94,7 @@ def _init_model_output(self, t):
             t = affine(t, 1, input_size=self.input_layer_sz_,
                        scope='output_layer', sparse_input=True)
         else:
-            t = tf.nn.dropout(t, keep_prob=self._dropout)
+            t = tf.nn.dropout(t, keep_prob=self._keep_prob)
```
Comment (beckermr) [inline muffnn/mlp/mlp_regressor.py]: Ditto from above.

File: muffnn/autoencoder/autoencoder.py (hunk 2 above)
Comment (beckermr) [inline muffnn/autoencoder/autoencoder.py]: Yep.

**Comment by beckermr** [comment]
If you are done, LGTM!
